core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -266,6 +266,23 @@ private static String toSql(RelNode root, SqlDialect dialect,
         .withInformix().ok(expectedInformix);
   }
 
+  /** Test case for <a href="https://issues.apache.org/jira/browse/CALCITE-6910">[CALCITE-6910]
+   * RelToSql does not handle ASOF joins</a>. */
+  @Test void testAsofJoin() {
+    final String sql = "select \"employee\".\"full_name\" from \"employee\" asof join \"product\"\n"
+        + "match_condition \"employee\".\"department_id\" <= \"product\".\"product_id\"\n"
+        + "on \"employee\".\"salary\" = \"product\".\"gross_weight\"";
+    final String expected = "SELECT \"t\".\"full_name\"\nFROM "
+        + "((SELECT \"employee_id\", \"full_name\", \"first_name\", \"last_name\", \"position_id\", "
+        + "\"position_title\", \"store_id\", \"department_id\", \"birth_date\", \"hire_date\", "
+        + "\"end_date\", \"salary\", \"supervisor_id\", \"education_level\", \"marital_status\", "
+        + "\"gender\", \"management_role\", CAST(\"salary\" AS DOUBLE) AS \"salary0\"\n"
+        + "FROM \"foodmart\".\"employee\") AS \"t\" ASOF JOIN \"foodmart\".\"product\" "
+        + "MATCH_CONDITION \"t\".\"department_id\" <= \"product\".\"product_id\" ON "
```

Comment:
According to the description of JoinRelType#ASOF, the match_condition should be followed by parentheses. However, the current generated SQL does not include parentheses. Does this need to be modified? Or is it indicating match_condition optional?

Reply:
You can see in the SQL compiled that even the input query does not require parentheses. I also see that there is a typo in the documentation.

Reply:
I have fixed the documentation in a separate PR